shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,374 @@
+  if (!fl_method_call_respond(method_call, response, &error)) {
+    g_warning("Failed to send method call response: %s", error->message);
+  }
+}
+
+// Creates a new platform view.
+static FlMethodResponse* platform_views_create(FlPlatformViewsPlugin* self,
+                                               FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_MAP) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  FlValue* id_value = fl_value_lookup_string(args, "id");
+  if (id_value == nullptr || fl_value_get_type(id_value) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view id", nullptr));
+  }
+  int64_t id = fl_value_get_int(id_value);
+
+  FlValue* view_type_value = fl_value_lookup_string(args, "viewType");
+  if (view_type_value == nullptr ||
+      fl_value_get_type(view_type_value) != FL_VALUE_TYPE_STRING) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view viewType", nullptr));
+  }
+  const gchar* view_type = fl_value_get_string(view_type_value);
+
+  FlValue* direction_value = fl_value_lookup_string(args, "direction");
+  if (direction_value == nullptr ||
+      fl_value_get_type(direction_value) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Missing platform view direction", nullptr));
+  }
+  GtkTextDirection direction =
+      static_cast<GtkTextDirection>(fl_value_get_int(direction_value));
+
+  FlValue* params_value = fl_value_lookup_string(args, "params");
+
+  FlPlatformView* platform_view =
+      fl_platform_views_plugin_get_platform_view(self, id);
+  if (platform_view) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view id already exists", nullptr));
+  }
+
+  FlPlatformViewFactory* factory =
+      FL_PLATFORM_VIEW_FACTORY(g_hash_table_lookup(self->factories, view_type));
+  if (!factory) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "View type does not exist", nullptr));
+  }
+
+  g_autoptr(FlMessageCodec) codec =
+      fl_platform_view_factory_get_create_arguments_codec(factory);
+  FlValue* creation_params = nullptr;
+  if (codec && params_value) {
+    const uint8_t* creation_params_bytes =
+        fl_value_get_uint8_list(params_value);
+    size_t creation_params_length = fl_value_get_length(params_value);
+    g_autoptr(GBytes) bytes =
+        g_bytes_new_static(creation_params_bytes, creation_params_length);
+
+    GError* error = nullptr;
+    creation_params = fl_message_codec_decode_message(codec, bytes, &error);
+  }
+
+  platform_view = fl_platform_view_factory_create_platform_view(
+      factory, id, creation_params);
+  if (!FL_IS_PLATFORM_VIEW(platform_view)) {
```

Comment:
Just check if `platform_view == nullptr`. The current code would leak the object if it wasn't. If it's not the correct object we'll hit assertions later that will indicate the rare case that some other object was passed through.